src/main/java/io/minio/MinioClient.java

```diff
@@ -152,7 +152,7 @@
    * @see #MinioClient(String endpoint, int port, String accessKey, String secretKey, boolean secure)
    */
   public MinioClient(String endpoint) throws InvalidEndpointException, InvalidPortException {
-    this(endpoint, 0, null, null, true);
+    this(endpoint, 0, null, null, endpoint==null?false:endpoint.startsWith("https://"));
```

Comment:
Ah, ok, that means in the end, that even if i specify "http://localhost" as the endpoint, i end up with an https connection? I think that is confusing, bu it is your api. And as long as your suggested changes makes it possible to get an connection when specifying a valid endpoint and a port number which really gets to the given port number and not to the default i am fine. 

But what about a policy to handle back https in any case where the endpoint does not explicitly specify something else? then 
`endpoint==null?true:endpoint.startsWith("http://")` would work just fine?